### DISCUSSION TITLE
Tab-out plugin default key <Tab> being overwritten by LazyVim

### DISCUSSION THREAD

**Discussion by speri203** [LazyVim/LazyVim#1924] [Q&A] [answered]
I have been trying to incorporate tab-out plugin (https://github.com/abecodes/tabout.nvim) and while it works initially, by the time the workspace its functionality completely breaks. When I load a file, for a few seconds, I am able to go in-between quotes and pressing `<Tab> ` goes to the closing quotes character. Afterwards the functionality of `<Tab>` is the default where it shifts content or adds the equivalent spaces of a tab. I tried doing `:verbose inmap <Tab>` and it points to the `coding.lua` file in LazyVim. Exploring this file I found that LuaSnip also utilizes the `<Tab>` key to cycle through suggestions. I have set the `keys = nil` and also tried setting it to an empty table (`keys = {}`)but the issue still persists. Is there nothing plugin that is overriding tab-out's functionality and how do I fix it? The config is listed below:

```
return {
  {
    "L3MON4D3/LuaSnip",
    keys = nil,
  },
  {
    'abecodes/tabout.nvim',
    event = "VeryLazy",
    config = function()
      require('tabout').setup {
        tabkey = '<Tab>',             -- key to trigger tabout, set to an empty string to disable
        backwards_tabkey = '<S-Tab>', -- key to trigger backwards tabout, set to an empty string to disable
        act_as_tab = true,            -- shift content if tab out is not possible
        act_as_shift_tab = false,     -- reverse shift content if tab out is not possible (if your keyboard/terminal supports <S-Tab>)
        default_tab = '<C-t>',        -- shift default action (only at the beginning of a line, otherwise <TAB> is used)
        default_shift_tab = '<C-d>',  -- reverse shift default action,
        enable_backwards = true,      -- well ...
        completion = true,            -- if the tabkey is used in a completion pum
        tabouts = {
          { open = "'", close = "'" },
          { open = '"', close = '"' },
          { open = '`', close = '`' },
          { open = '(', close = ')' },
          { open = '[', close = ']' },
          { open = '{', close = '}' }
        },
        ignore_beginning = true, --[[ if the cursor is at the beginning of a filled element it will rather tab out than shift the content ]]
        exclude = {} -- tabout will ignore these filetypes
      }
    end,
  }
}
```

**Comment by dpetka2001** [comment, marked as answer]
In the [examples](https://www.lazyvim.org/configuration/examples) docs towards the end of the file, you can see how to overwrite the Luasnip default `keys` section. You should do
```lua
{
    "L3MON4D3/LuaSnip",
    keys = function()
      return {}
    end,
  },
```

**Reply by speri203** [reply]
Ahh thank you, I missed that portion of the docs. The issue is solved!
